### PR TITLE
Object Oriented Programming

### PR DESCRIPTION
**Pull request by G1Joshi**

### CODE REVIEW

File: OOP/Object Oriented Programming/README.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+Destructors are special functions, as well. They’re called when an object is destroyed or deleted.
+
+
+## Friend Functions
+Normally, private members of a class cannot be accessed from outside of that class.
+However, declaring a non-member function as a friend of a class allows it to access the class’ private members. This is accomplished by including a declaration of this external function within the class, and preceding it with the keyword friend.
+Typical use cases of friend functions are operations that are conducted between two different classes accessing private members of both.
+
+### This
+Every object in C++ has access to its own address through an important pointer called the this pointer.
```
Comment (gcnit) [inline OOP/Object Oriented Programming/README.md:91]: @G1Joshi please avoid sentences taken from other websites.

Comment (G1Joshi) [reply OOP/Object Oriented Programming/README.md:91]: These are my notes made by me during my academics.

Comment (gcnit) [reply OOP/Object Oriented Programming/README.md:91]: https://www.tutorialspoint.com/cplusplus/cpp_this_pointer.htm#:~:text=Every%20object%20in%20C%2B%2B%20has,refer%20to%20the%20invoking%20object.

Comment (G1Joshi) [reply OOP/Object Oriented Programming/README.md:91]: Yes, I also take some help from other resources, but I am not sure that they are ditto same.

**Pull request closed without merging**